# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1319771795 25200
#      Thu Oct 27 20:16:35 2011 -0700
# Node ID e9c23efe6ffffee6a4ebbda08e8f9766afa89d74
# Parent  85d63b6e49cfa033e1e7cbddc910a0a65f1df58e
wavwrite.m: Use temporary filename in %!tests rather than hardcoded one.

* wavwrite.m: Use temporary filename in %!tests rather than hardcoded one.

diff --git a/scripts/audio/wavwrite.m b/scripts/audio/wavwrite.m
--- a/scripts/audio/wavwrite.m
+++ b/scripts/audio/wavwrite.m
@@ -138,43 +138,46 @@ function wavwrite (y, varargin)
   ## write to file
   c = fwrite (fid, yi, format, 0, BYTEORDER);
 
   fclose (fid);
 
 endfunction
 
 
+%!shared fname
+%! fname = tmpnam ();
+
 %!test
 %! A = [-1:0.1:1; -1:0.1:1];
-%! wavwrite (A, "a.wav");
-%! [B, samples_per_sec, bits_per_sample] = wavread ("a.wav");
-%! assert(A,B, 1/2^15);
-%! assert(samples_per_sec, 8000);
-%! assert(bits_per_sample, 16);
-%! delete ("a.wav");
+%! wavwrite (A, fname);
+%! [B, samples_per_sec, bits_per_sample] = wavread (fname);
+%! assert (A,B, 1/2^15);
+%! assert (samples_per_sec, 8000);
+%! assert (bits_per_sample, 16);
+%! unlink (fname);
 %
 %!test
 %! A = [-1:0.1:1; -1:0.1:1];
-%! wavwrite (A, 4000, "a.wav");
-%! [B, samples_per_sec, bits_per_sample] = wavread ("a.wav");
-%! assert(A,B, 1/2^15);
-%! assert(samples_per_sec, 4000);
-%! assert(bits_per_sample, 16);
-%! delete ("a.wav");
+%! wavwrite (A, 4000, fname);
+%! [B, samples_per_sec, bits_per_sample] = wavread (fname);
+%! assert (A,B, 1/2^15);
+%! assert (samples_per_sec, 4000);
+%! assert (bits_per_sample, 16);
+%! unlink (fname);
 %
 %!test
 %! A = [-1:0.1:1; -1:0.1:1];
-%! wavwrite (A, 4000, 8, "a.wav");
-%! [B, samples_per_sec, bits_per_sample] = wavread ("a.wav");
-%! assert(A,B, 1/128);
-%! assert(samples_per_sec, 4000);
-%! assert(bits_per_sample, 8);
-%! delete ("a.wav");
+%! wavwrite (A, 4000, 8, fname);
+%! [B, samples_per_sec, bits_per_sample] = wavread (fname);
+%! assert (A,B, 1/128);
+%! assert (samples_per_sec, 4000);
+%! assert (bits_per_sample, 8);
+%! unlink (fname);
 %
 %!test
 %! A = [-2:2];
-%! wavwrite (A, "a.wav");
-%! B = wavread ("a.wav");
+%! wavwrite (A, fname);
+%! B = wavread (fname);
 %! B *= 32768;
-%! assert(B, [-32768 -32768 0 32767 32767]);
-%! delete ("a.wav");
+%! assert (B, [-32768 -32768 0 32767 32767]);
+%! unlink (fname);
 
